### PR TITLE
Added instructions for getting stack traces with gdb

### PR DESCRIPTION
**Pull request by NanoAkron**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -385,6 +385,20 @@ If you want to help out, see CONTRIBUTING for a set of guidelines.
 
 This section contains general instructions for debugging failed installs or problems encountered with Monero. First ensure you are running the latest version built from the github repo.
 
+## Obtaining Stack Traces in Linux
+
+We generally use the tool `gdb` to provide stack trace functionality, althout this is currently being integrated into the easylogger++ system by @moneromooo-monero to provide stack traces across all systems
```
Comment (moneromooo-monero) [inline README.md]: gdb is for post mortem stack traces, or instantaneous ones.
easylogging++ is for programming logging of stack traces, including on exceptions. It won't replace using gdb.

File: README.md [hunk 2]
```diff
@@ -385,6 +385,20 @@ If you want to help out, see CONTRIBUTING for a set of guidelines.
...
+
+To use gdb:
+
+``
+gdb /path/to/monerod `pidof monerod`
```
Comment (moneromooo-monero) [inline README.md]: This is when getting a trace from a running process. To get a trace for a dead process: gdb /path/to/monerod /path/to/core/file  (and you need to have a core file, so you may need to run "ulimit -c unlimited" first, if it's set to 0)

File: README.md (hunk 1 above)
Comment (NanoAkron) [inline README.md]: Ah

File: README.md (hunk 2 above)
Comment (NanoAkron) [inline README.md]: Do you run the "ulimit -c unlimited" commands inside gdb?

**Comment by NanoAkron** [comment]
Do you run the "ulimit -c unlimited" commands inside gdb?

**Comment by hyc** [comment]
No, ulimit is a shell command not a gdb command.

**Comment by moneromooo-monero** [comment]
And you run it before running the process that might crash.
You don't need it to run a process which you want to gather a runtime trace. Just the ones you want to get a core from.

**Comment by kenshi84** [comment]
Can I do the same for Mac? I guess I need to use `lldb` instead of `gdb`, but I get the following response.

    lldb /Users/kenshi/monero/monerod `pidof monerod`
    -bash: pidof: command not found`

Also, does the binary need to be the debug build?

**Comment by hyc** [comment]
to get the PID of monerod, you could instead use
```ps -ac | grep monerod | awk '{print $1}'```

**Comment by hyc** [comment]
And yes, generally you should use a debug build. A release build will be missing a lot of details.

**Comment by NanoAkron** [comment]
@hyc how should I amend for mac?

**Comment by kenshi84** [comment]
@hyc Thanks a lot! I was stupid enough to take `` `pidof monerod` `` as the exact command to be typed:) So this means I need to run `monerod` first before running `lldb`, correct? I'm still confused by the response:

    $ ps -ac | grep monerod | awk '{print $1}'
    44813
    $ lldb /Users/kenshi/monero/monerod 44813
    (lldb) target create "/Users/kenshi/monero/monerod"
    Current executable set to '/Users/kenshi/monero/monerod' (x86_64).
    (lldb) settings set -- target.run-args  "44813"
    (lldb) run
    Process 44831 launched: '/Users/kenshi/monero/monerod' (x86_64)
    Monero 'Wolfram Warptangent' (v0.10.1.0-4f06ba6)
    Commands: 
      alt_chain_info          Print information about alternative chains
      ban                     Ban a given IP for a time
      bans                    Show the currently banned IPs
      bc_dyn_stats            Print information about current blockchain dynamic state
      diff                    Show difficulty
      exit                    Stop the daemon
      flush_txpool            Flush a transaction from the tx pool by its txid, or the whole tx pool
      hard_fork_info          Print hard fork voting information
      help                    Show this help
      hide_hr                 Stop showing hash rate
      is_key_image_spent      Prints whether a given key image is in the spent key images set, is_key_image_spent <key_image>
      limit                   limit <kB/s> - Set download and upload limit
      limit_down              limit <kB/s> - Set download limit
      limit_up                limit <kB/s> - Set upload limit
      out_peers               Set max number of out peers
      output_histogram        Print output histogram (amount, instances)
      print_bc                Print blockchain info in a given blocks range, print_bc <begin_height> [<end_height>]
      print_block             Print block, print_block <block_hash> | <block_height>
      print_cn                Print connections
      print_coinbase_tx_sum   Print sum of coinbase transactions (start height, block count)
      print_height            Print local blockchain height
      print_pl                Print peer list
      print_pool              Print transaction pool (long format)
      print_pool_sh           Print transaction pool (short format)
      print_pool_stats        Print transaction pool statistics
      print_status            Prints daemon status
      print_tx                Print transaction, print_tx <transaction_hash>
      q                       ignored
      save                    Save blockchain
      set_log                 set_log <level>|<categories> - Change current loglevel, <level> is a number 0-4
      show_hr                 Start showing hash rate
      start_mining            Start mining for specified address, start_mining <addr> [<threads>], default 1 thread
      start_save_graph        Start save data for dr monero
      status                  Show status
      stop_daemon             Stop the daemon
      stop_mining             Stop mining
      stop_save_graph         Stop save data for dr monero
      unban                   Unban a given IP
          
    
    Unknown command
    Process 44831 exited with status = 1 (0x00000001) 

If I do `lldb /Users/kenshi/monero/monerod` instead, I get the following response:

    $ lldb /Users/kenshi/monero/monerod
    (lldb) target create "/Users/kenshi/monero/monerod"
    Current executable set to '/Users/kenshi/monero/monerod' (x86_64).
    (lldb) run
    Process 45047 launched: '/Users/kenshi/monero/monerod' (x86_64)
    2017-01-23 08:36:43.571   INFO  global  src/daemon/main.cpp:265 Monero 'Wolfram Warptangent' (v0.10.1.0-4f06ba6)
    2017-01-23 08:36:43.572   INFO  global  src/daemon/protocol.h:53  Initializing cryptonote protocol...
    2017-01-23 08:36:43.572   INFO  global  src/daemon/protocol.h:58  Cryptonote protocol initialized OK
    2017-01-23 08:36:43.572   INFO  global  src/daemon/p2p.h:63 Initializing p2p server...

Which form is correct?

Also I've had no success in building the code in the debug mode on Mac due to strange link errors with boost. Has anyone succeeded in it?

**Comment by moneromooo-monero** [comment]
You're trying to run monerod in gdb here. Different thing.

Run gdb on a running process: gdb monerod `pidof monerod`
Run gdb on a core: gdb monerod core
Run gdb on a new monerod: gdb monerod, then "run" in gdb

**Comment by NanoAkron** [comment]
Will change

**Comment by kenshi84** [comment]
@moneromooo-monero Thanks and sorry for being such a noob. Previously I was mainly developing on Windows using Visual Studio and didn't need to learn how to use gdb/lldb. I guess I should first educate myself for a bit.

Still I'd be curious if any Mac users have successfully built monero in the debug mode.

**Pull request merged**